### DISCUSSION TITLE
Units of Injection current, shift current, Optical conductivity

### DISCUSSION THREAD

**Discussion by yilinzhao222** [wannier-berri/wannier-berri#293] [Q&A]
Dear all,

I am kind of confused about the units of the properties calculated in wb. I checked all of the documents, but I could not find any related docs which describes the units of the physical quantities calculated with wb. Could anyone help me figure out the units of Injection current, shift current, optical conductivity in wb please?

Regards,
Frank

**Comment by stepan-tsirkin** [comment]
@yilinzhao222 

The units of all outputs after integration in WannierBerri are the SI units, to avoid confusion. So, for optical conductivity it is Siemens per meter (S/m) 

Similarly, for other quantities just figure out the corresponding SI units. 

also note that injection current is proportional to the relaxation time $\tau$ which is arbitrarily taken as 1 fs

https://github.com/wannier-berri/wannier-berri/blob/91e5b6983bd82f6017877271e222d4030cf928b4/wannierberri/__factors.py#L3-L4

https://github.com/wannier-berri/wannier-berri/blob/91e5b6983bd82f6017877271e222d4030cf928b4/wannierberri/__factors.py#L33-L34

so, you have to multipl;y the result by the value of $\tau$ that seems reasonable for your material.

**Reply by yilinzhao222** [reply]
Hi Stepan,

Thank you for your response. It is really helpful. For the calculation of injection current, I have another question. When dealing with big system, the number of bands are usually very large. I noticed that for the energy or berry calculation, we can choose which bands are included with ibands. Could we choose which bands (especially those near the Fermi level) are included during the calculation in the same way?  

Regards,
Frank

**Reply by stepan-tsirkin** [reply]
good question. I remember I had smth like that in mind when I set  parameters `Emin` and `Emax` in data_K

https://github.com/wannier-berri/wannier-berri/blob/91e5b6983bd82f6017877271e222d4030cf928b4/wannierberri/data_K.py#L74-L75

Those parameters are passed via the `parameters_K` dictionary in the `run()` input.

https://github.com/wannier-berri/wannier-berri/blob/91e5b6983bd82f6017877271e222d4030cf928b4/wannierberri/run.py#L112

But I am not sure if it still works and if it gives notable performance improvement.
